runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -647,7 +647,7 @@ void killed(int amt)
 		// save changed value
 		setProfileConfig(SlayerConfig.AMOUNT_KEY, amount);
 
-		if (!config.showInfobox())
+		if (!config.showInfobox() || Task.getTask(taskName) == null)
```

Comment:
This should be `Strings.isNullOrEmpty(taskName)` instead, since tracking unknown tasks is still valid and something the plugin will do. Even better would be to just nullcheck `counter` inside of `killed` though.

Reply:
It's still unclear how we get to `counter.setCount(amount)` with an empty taskName while amount > 0. It seems it should be impossible to get to this line with an empty taskName but also a non zero amount since we set both through setTask.

If there is an edge case where we're not finding the slayer task name at all in the chat box for whatever reason, maybe it would be even better to just force it to something like "Unknown task" when we find taskName is null or empty? We already support unknwon task names as you said, so that would probably cover this case too.

Reply:
Oh I had not seen the `amount` check, so I agree I do not see how this can be happening then. Maybe we wait and see if someone figures out how to reproduce the issue.

Reply:
There are a couple of suspicious looking writes to amount, like for bracelet of slaughters. You might be able to 1) get a task on mobile 2) proc a slaughter bracelet to set amount = 1 3) get an xp drop - which would then trip it?